RPG Wizardry/Assets/Scripts/Gameplay/Projectile.cs

```diff
@@ -47,9 +47,10 @@ protected virtual void Move()
     /// Applies the spell's effect to the colliding object (usually damage)
     /// </summary>
     /// <param name="collision"></param>
-    protected virtual void Effect(Collision collision)
+    protected virtual void Effect(Collider2D collision)
     {
         //oh man i can feel the effect
+        Destroy(gameObject);
```

Comment:
Move this line to the OnTriggerEnter2D-Method, as it's not part of the Effect.

Reply:
I have it there in case we later have spells that don't immediately destroy self on collision, like something that bounces or is a constant beam